pkg/elb/cleanup.go

```diff
@@ -199,6 +199,26 @@ func getServiceLoadBalancer(ctx context.Context, ec2API awsapi.EC2, elbAPI Descr
 	return &lb, nil
 }
 
+func getIngressELBName(ctx context.Context, metadataName string, hosts []string) string {
+	// Expected e.g. bf647c9e-default-appingres-350b-1622159649.eu-central-1.elb.amazonaws.com where AWS ALB name is
+	// bf647c9e-default-appingres-350b (cannot be longer than 32 characters).
+	hostNameParts := strings.Split(hosts[0], ".")
+	if len(hostNameParts[0]) == 0 {
+		logger.Debug("%s is ALB Ingress, but probably not provisioned or something other unexpected, skip", metadataName)
+		return ""
+	}
+	name := strings.TrimPrefix(hostNameParts[0], "internal-") // Trim 'internal-' prefix for ALB DNS name which is not a part of name.
+	idIdx := strings.LastIndex(name, "-")
+	if (idIdx) != -1 {
+		name = name[:idIdx] // Remove the ELB ID and last hyphen at the end of the hostname (ELB name cannot end with a hyphen)
+	}
+	if len(name) > 32 {
+		name = name[:32]
+	}
+	logger.Debug("ALB resource name: %s", name)
```

Comment:
Thanks for the suggestion! But I am not sure if this update is what you are looking for? Or does it mean the debug message of showing ALB resource name should be moved to `getIngressLoadBalancer`?

- https://github.com/eksctl-io/eksctl/pull/7029/commits/4989bf86801145276785f5691c310e808f33cf12#diff-bf8761431813288f0320030b76870e9d4a40dd0e80e1ca725ef262184c498b90R217

I did not fully understand what could be the difference as it seems like not all logger in the package is applying the defer directive, I assume it might be some performance improvement, or, it helps the readability (which I think would be what you are mentioning). Any details would be appreciated!

Reply:
Moved the logger to `getIngressLoadBalancer`: https://github.com/eksctl-io/eksctl/compare/4989bf86801145276785f5691c310e808f33cf12..b73c890febc5feec8b4ae554067260aecfe73a91